pkg/shell/cockpit-server.js

```diff
@@ -581,11 +571,15 @@ PageShutdownDialog.prototype = {
         else
             when = "+" + delay;
 
-        this.cockpitd_manager.call('Shutdown', op, when, message, function(error) {
-            $('#shutdown-dialog').modal('hide');
-            if (error && error.name != 'Disconnected')
-                shell.show_unexpected_error(error);
-        });
+        var arg = (op == "shutdown") ? "--poweroff" : "--reboot";
+        cockpit.spawn(["shutdown", arg, when, message])
```

Comment:
> Still don't know why the test succeeds...

Heh, it doesn't.  It waits for a reboot that never comes.  I thought I had seen a test run that failed in check-docker, but that was probably some other pull request.

Ok, it all makes sense again.  I'll add `superuser: true`, test, and merge.